tests/network/firewalld_policy_objects.pm

```diff
@@ -0,0 +1,212 @@
+    if ($result ne $expected) {
+        die "Result of last command is: $result. I was expecting: $expected";
+    }
+}
+
+sub connection_test {
+    # Tests connection from Client to Server machines and vice-versa. Http GET and ping are used.
+    # Results are compared with given expected ones
+    my ($hostname, $client_wget_result, $client_ping_result, $server_wget_result, $server_ping_result) = @_;
+    my $result = "100";
+
+    if ($hostname eq "client") {
+        record_info 'Client to Server connection test';
+        $result = script_run("wget $SRV_IP:$SRV_PORT/hostedfile.txt");
+        check_result($result, $client_wget_result);
+
+        $result = script_run("ping -c 3 $SRV_IP");
+        check_result($result, $client_ping_result);
+
+    } elsif ($hostname eq "server") {
+        record_info 'Server to Client connection test';
+        $result = script_run("wget $CLI_IP:$SRV_PORT/hostedfile.txt");
+        check_result($result, $server_wget_result);
+
+        $result = script_run("ping -c 3 $CLI_IP");
+        check_result($result, $server_ping_result);
+    }
+}
+
+sub configure_firewall_policies {
+    # Create new zones and policies for the test
+    my ($net0, $net1) = @_;
+    assert_script_run("systemctl start firewalld");
+    assert_script_run("firewall-cmd --permanent --new-zone=my-external");
+    assert_script_run("firewall-cmd --permanent --new-zone=my-internal");
+    assert_script_run("systemctl restart firewalld");
+    assert_script_run("firewall-cmd --zone=my-external --change-interface=$net0");
+    assert_script_run("firewall-cmd --zone=my-internal --change-interface=$net1");
+    assert_script_run("firewall-cmd --permanent --new-policy=int-to-ext");
+    assert_script_run("firewall-cmd --permanent --new-policy=ext-to-int");
+
+    assert_script_run("firewall-cmd --permanent --policy int-to-ext --add-ingress-zone=my-internal");
+    assert_script_run("firewall-cmd --permanent --policy int-to-ext --add-egress-zone=my-external");
+    assert_script_run("firewall-cmd --permanent --policy ext-to-int --add-ingress-zone=my-external");
+    assert_script_run("firewall-cmd --permanent --policy ext-to-int --add-egress-zone=my-internal");
+    assert_script_run("firewall-cmd --runtime-to-permanent");
+
+    # Internal to External policy: Allow http and icmp
+    assert_script_run("firewall-cmd --permanent --policy int-to-ext --add-rich-rule='rule family=ipv4 service name=http accept'");
+    assert_script_run("firewall-cmd --permanent --policy int-to-ext --add-rich-rule='rule family=ipv4 protocol value=icmp accept'");
+    # External to Internal policy: Reject http and icmp
+    assert_script_run("firewall-cmd --permanent --policy ext-to-int --add-rich-rule='rule family=ipv4 service name=http reject'");
+    assert_script_run("firewall-cmd --permanent --policy ext-to-int --add-rich-rule='rule family=ipv4 protocol value=icmp reject'");
+
+    assert_script_run("systemctl restart firewalld");
```

Comment:
isn't it enough to `firewall-cmd --reload` ?
Either way LGTM! good stuff

Reply:
You are right, thanks!